src/ceph-disk/ceph_disk/main.py

```diff
@@ -2625,8 +2625,11 @@ def __init__(self, args):
 
     def set_type(self):
         dmode = os.stat(self.args.data).st_mode
+        realPath = os.path.realpath(self.args.data)
 
-        if stat.S_ISDIR(dmode):
+        if realPath[:5] == '/dev/' and not stat.S_ISBLK(dmode):
```

Comment:
also the commit message does not read right

> prepare accepts a directory who's name starts with /dev

it actually **rejects** a directory whose name starts with "/dev/"